api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -147,4 +150,37 @@ public String generateVisitNumber(Location location, Patient patient) {
         return dateString + "-" + locationName + "-" + zeroesToAppend + nextNumberInQueue;
     }
 
+    /**
+     * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueListBySearchParams(java.lang.String, java.util.Date, java.util.Date, org.openmrs.Location, org.openmrs.Location, org.openmrs.module.patientqueueing.model.PatientQueue.Status)
+     */
+    @Override
+    public List<PatientQueue> getPatientQueueListBySearchParams(String searchString, Date fromDate, Date toDate, Location locationTo, Location locationFrom, PatientQueue.Status status) {
+        if (fromDate == null || toDate == null) {
+            fromDate = OpenmrsUtil.firstSecondOfDay(new Date());
+            toDate = OpenmrsUtil.getLastMomentOfDay(new Date());
+        }
+        return dao.getPatientQueueList(getPatientByNames(searchString), fromDate, toDate, locationTo, locationFrom, status);
+    }
+
+    /**
+     * Get list of patients that match search string
+     *
+     * @param searchString This is the search string that will be used to get patients in a list
+     * @return List of patients that match the search criteria
+     */
+    private List<Patient> getPatientByNames(String searchString) {
```

Comment:
Doesn't openmrs already have an api of getting patients by name?

Reply:
@dkayiwa search the etire openmrs repo but in vain